ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/TableSelectBuilder.java

```diff
@@ -56,20 +69,71 @@ private TableSelectBuilder() {
 
     final Named selectName = Named.as(StreamsUtil.buildOpName(queryContext));
 
+    final Optional<MaterializationInfo.Builder> matBuilder =  table.getMaterializationBuilder();
+
+    final boolean forceMaterialize = !matBuilder.isPresent();
+
+    //Feauture flag for turning T-T Join queryability on or off
+    final boolean turnItOn = true;
```

Comment:
Seems like this could have a better name ;) 

I.e., the name should say what the comment says. Then, you don't need the comment.